File: Angular_Exam_Portal/app/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseurl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private http:HttpClient) { }

  public quizzes(){
    return this.http.get(`${baseurl}/quiz/`)
  }
  public addQuiz(quiz:any){
    return this.http.post(`${baseurl}/quiz/`,quiz);

  }
  public deleteQuiz(qid:any){
    return this.http.delete(`${baseurl}/quiz/${qid}`);
  }
  public getQuiz(qid:any){
    return this.http.get(`${baseurl}/quiz/${qid}`);
  }

  public updateQuiz(quiz: any){
    return this.http.put(`${baseurl}/quiz/`,quiz);
  }

  public getQuizzesOfCategory(cid:any){
    return this.http.get(`${baseurl}/quiz/category/${cid}`);
  }
  //get active quizzess
  public getActiveQuizzess(){
    return this.http.get(`${baseurl}/quiz/active`);
  }

  //get active quizzess of category
  public getActiveQuizzesOfCategory(cid:any){
    return this.http.get(`${baseurl}/quiz/category/active/${cid}`);
  }
}

File: SpringBoot_Backend_Exam_portal/src/main/java/com/exam/services/QuizService.java
package com.exam.services;

import java.util.List;
import java.util.Set;

import org.springframework.http.ResponseEntity;

import com.exam.entity.Category;
import com.exam.entity.Quiz;

public interface QuizService {
	
	public Quiz addQuiz(Quiz quiz);
	public Quiz updateQuiz(Quiz quiz);
	public Set<Quiz> getQuizzes();
	public Quiz getQuiz(Long quizId);
	public void deleteQuiz(Long quizId);
	public List<Quiz> getQuizzesOfCategory(Category c);
	public List<Quiz> getActiveQuizzes();
	public List<Quiz> getActiveQuizzesOfCategory(Category c);
}
